### PR TITLE
Document the process for releasing a new sidecar

### PR DESCRIPTION
**Pull request by msau42**
Fixes: https://github.com/kubernetes-csi/docs/issues/142

### CODE REVIEW

File: SIDECAR_RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+The release manager must:
+
+* Be a member of the kubernetes-csi organization. Open an
+  [issue](https://github.com/kubernetes/org/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.md&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E) in
+  kubernetes/org to request membership
+* Be a top level approver for the repository. To become a top level approver,
+  the candidate must demonstrate ownership and deep knowledge of the repository
+  through active maintainence, responding to and fixing issues, reviewing PRs,
+  test triage.
+* Be part of the maintainers group for the repository. Membership can be
```
Comment (xing-yang) [inline SIDECAR_RELEASE_PROCESS.md]: Do you have to be in the maintainers group to cut a release?  I'm not in external-snapshotter-maintainers.  Looks like I should apply for it:).

File: SIDECAR_RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+## Updating CI Jobs
+Whenever a new Kubernetes minor version is released, our kubernetes-csi CI jobs
+must be updated.
+
+[Our CI jobs](https://k8s-testgrid.appspot.com/sig-storage-csi-ci) have the
+naming convention `<hostpath-deployment-version>-on-<kubernetes-version>`.
+
+1. Jobs should be actively monitored to find and fix failures in sidecars and
+   infrastructure changes early in the development cycle.
+1. "-on-master" jobs are the closest reflection to the new Kubernetes version.
```
Comment (xing-yang) [inline SIDECAR_RELEASE_PROCESS.md]: Why are these all "1."?

File: SIDECAR_RELEASE_PROCESS.md (hunk 1 above)
Comment (msau42) [reply SIDECAR_RELEASE_PROCESS.md]: or admin group. will update

File: SIDECAR_RELEASE_PROCESS.md (hunk 2 above)
Comment (msau42) [reply SIDECAR_RELEASE_PROCESS.md]: markdown automatically orders them. that way if you add something in the middle, you don't have to renumber all the rest

File: SIDECAR_RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+   is released, we can make the optional jobs required, and also remove the
+   Kubernetes versions that are no longer supported.
+
+## Release Process
+1. Identify all issues and ongoing PRs that should go into the release, and
+  drive them to resolution.
+1. Download [K8s release notes
+  generator](https://github.com/kubernetes/release/tree/master/cmd/release-notes)
+1. Generate release notes for the release. Replace arguments with the relevant
+  information.
```
Comment (jsafrane) [inline SIDECAR_RELEASE_PROCESS.md]: For discussion:
`--start-sha=` points to the last release from the same branch. E.g.:
* `1.X-1.0` tag when releasing `1.X.0`.
* `1.X.Y-1` when releasing `1.X.Y`.

This way, CHANGELOG-1.3 may repeat fixes in master that were already backported to release-1.2 and mentioned in CHANGELOG-1.2 in say 1.2.2 release.

File: SIDECAR_RELEASE_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+   and any other sections that may need updating.
+1. Check that all [canary CI
+  jobs](https://k8s-testgrid.appspot.com/sig-storage-csi-ci) are passing,
+  and that test coverage is adequate for the changes that are going into the release.
+1. Make sure that no new PRs have merged in the meantime, and no PRs are in
+   flight and soon to be merged.
+1. [Create a new
+   release](https://github.com/kubernetes-csi/external-provisioner/releases/new)
+   following a previous release as a template. Be sure to select the correct
+   branch.
```
Comment (jsafrane) [inline SIDECAR_RELEASE_PROCESS.md]: `Create a new release ([external-provisioner example](https://github.com/kubernetes-csi/external-provisioner/releases/new))...`

So it's clear the link does not release all sidecars.

File: SIDECAR_RELEASE_PROCESS.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+## Release Process
+1. Identify all issues and ongoing PRs that should go into the release, and
+  drive them to resolution.
+1. Download [K8s release notes
+  generator](https://github.com/kubernetes/release/tree/master/cmd/release-notes)
+1. Generate release notes for the release. Replace arguments with the relevant
+  information.
+    ```
+    ./release-notes --start-sha=0ed6978fd199e3ca10326b82b4b8b8e916211c9b --end-sha=3cb3d2f18ed8cb40371c6d8886edcabd1f27e7b9 \
```
Comment (pohly) [inline SIDECAR_RELEASE_PROCESS.md]: This needs a GitHub token in `GITHUB_TOKEN`, doesn't it?

https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line

File: SIDECAR_RELEASE_PROCESS.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    --github-org=kubernetes-csi --github-repo=external-attacher -branch=master -output out.md
+    ```
+1. Compare the generated output to the new commits for the release to check if
+   any notable change missed a release note.
+1. Reword release notes as needed. Make sure to check notes for breaking
+   changes and deprecations.
+1. If release is a new major/minor version, create a new `CHANGELOG-<minor>.md`
```
Comment (pohly) [inline SIDECAR_RELEASE_PROCESS.md]: `CHANGELOG-<major>.<minor>.md` may be clearer.

**Comment by jsafrane** [comment]
lgtm
@pohly, what do you think?

**Comment by xing-yang** [comment]
lgtm

File: SIDECAR_RELEASE_PROCESS.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+1. Submit a PR for the CHANGELOG changes.
+1. Submit a PR for README changes, in particular, Compatibility, Feature status,
+   and any other sections that may need updating.
+1. Check that all [canary CI
+  jobs](https://k8s-testgrid.appspot.com/sig-storage-csi-ci) are passing,
+  and that test coverage is adequate for the changes that are going into the release.
+1. Make sure that no new PRs have merged in the meantime, and no PRs are in
+   flight and soon to be merged.
+1. Create a new release following a previous release as a template. Be sure to select the correct
+   branch. [external-provisioner example](https://github.com/kubernetes-csi/external-provisioner/releases/new)
```
Comment (xing-yang) [inline SIDECAR_RELEASE_PROCESS.md]: For people who have no permissions to cut a release on external-provisioner repo, they'll just get "https://github.com" when they click on this link.

Comment (msau42) [reply SIDECAR_RELEASE_PROCESS.md]: is there something you want me to add to the doc about this?

Comment (xing-yang) [reply SIDECAR_RELEASE_PROCESS.md]: Maybe just list this link https://github.com/kubernetes-csi/external-provisioner/releases/new and add a note that you'll need permissions to really open this link.

Comment (msau42) [reply SIDECAR_RELEASE_PROCESS.md]: done

**Pull request merged**